packages/react/src/__tests__/ReactChildren-test.js

```diff
@@ -1039,6 +1039,32 @@ describe('ReactChildren', () => {
     });
   });
 
+  // @gate __DEV__
+  it('does not throw on children without `_store`', async () => {
+    function ComponentRenderingFlattenedChildren({children}) {
+      return <div>{React.Children.toArray(children)}</div>;
+    }
+
+    const source = <div />;
+    const productionElement = {};
+    Object.entries(source).forEach(([key, value]) => {
+      if (key !== '_owner' && key !== '_store') {
+        productionElement[key] = value;
+      }
+    });
```

Comment:
The way this test case needs to be contrived signals that this mix of environments should really not be supported. We could consider adding build types to the version checks to prevent apps from relying on these setups in the future.